apps/bookmark/js/bookmark_editor.js

```diff
@@ -70,6 +73,25 @@ var BookmarkEditor = {
     var icon = new Icon(this.bookmarkIcon, this.data.icon);
     icon.render({'size': this.BOOKMARK_ICON_SIZE});
   },
+
+  _renderAppIcon: function renderAppIcon(manifest, manifestURL, size) {
+    // Parse icon URL from app manifest
+    var iconURL = window.WebManifestHelper.iconURLForSize(manifest,
+      manifestURL, size);
+    if (!iconURL) {
+      return;
+    }
+    // Switch out placeholder for real icon when it loads
+    this.appIconListener = this._handleAppIconLoad.bind(this);
+    this.appIcon.addEventListener('load', this.appIconListener);
+    this.appIcon.setAttribute('src', iconURL);
```

Comment:
Oh, does the homescreen do that to app icons? I thought that was only for bookmarks?